shadow-dom/focus/text-selection-with-delegatesFocus.html

```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-actions.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+
+<body>
+    <x-shadow id="withoutFocus"></x-shadow>
+    <x-shadow id="withFocus"></x-shadow>
+</body>
+
+<script>
+'use strict';
+
+/**
+ * build shadow-root with delegates focus, a focusable element, and an (ideally) selectable text
+ */
+function buildShadowRootWithSelectableText( element, shouldDelegateFocus ) {
+    element.attachShadow({ mode: 'open', delegatesFocus: shouldDelegateFocus });
+    const span = document.createElement('span');
+    span.textContent = 'Example Text to Select ';
+    const button = document.createElement('button');
+    button.textContent = 'Button';
+    element.shadowRoot.append(span);
+    element.shadowRoot.append(button);
+}
+
+/**
+ * command to select text in shadow-root
+ */
+function selectText(element, start, end) {
+    getSelection().empty();
+    const actions = new test_driver.Actions();
+    actions.pointerMove(start, 0, {origin: element});
+    actions.pointerDown();
+    actions.pointerMove(end, 0, {origin: element});
+    actions.pointerUp();
+    return actions.send();
+}
+
+promise_test(async () => {
+    const xShadow = document.getElementById('withoutFocus');
+    buildShadowRootWithSelectableText(xShadow, false);
+    await selectText(xShadow, -40, 45)
```

Comment:
-40 and +45 seem rather arbitrary. Maybe just include a comment here that we're just trying to select across the shadow root?

Reply:
ah, yeah, I can add that! I can also make the selection more sane, maybe something like `(0, 40)`, and annotate that this will select from the center, and the important bit is that we start the selection from inside the element (and not outside, where the focus will not be delegated)